react/week4/git-api/src/components/UserContext.js

```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useEffect, useState } from "react"
+
+const API = `https://api.github.com/search/users?q=`
+
+export const UserContext = createContext()
+
+export const UserProvider = ({ children, input, loading, setLoading }) => {
+  const [users, setUsers] = useState([])
+
+  useEffect(() => {
+    if (input) {
+      setLoading(true)
+      fetch(`${API}${input}`)
+        .then(res => res.json())
+        .then(data => {
+          data.message
+            ? alert("API rate limit exeeded, please wait some seconds")
```

Comment:
An alternative solution to the rate limit issue is to fetch items only with the first character, then save that in a state, and when the user enters the next character, you just filter down the already fetched array of items. Then you will have to make a new fetch only when the user removes the word completely and starts writing a new word. 
This way you can reduce the number of fetches to a minimum level, and the user can search for items without alert message popping up.